awx_collection/plugins/modules/tower_user.py

```diff
@@ -114,48 +122,72 @@
 def main():
     # Any additional arguments that are not fields of the item can be added here
     argument_spec = dict(
-        username=dict(required=True),
-        first_name=dict(),
-        last_name=dict(),
-        password=dict(no_log=True),
-        email=dict(required=False, default=''),
-        superuser=dict(type='bool', default=False),
-        auditor=dict(type='bool', default=False),
+        username=dict(required=True, type='str'),
+        first_name=dict(required=False, type='str'),
+        last_name=dict(required=False, type='str'),
+        email=dict(required=False, type='str'),
+        is_superuser=dict(required=False, type='bool', default='False', aliases=['superuser']),
+        is_system_auditor=dict(required=False, type='bool', default='False', aliases=['auditor']),
+        password=dict(required=False, type='str', default=''),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
+# TODO: Implement a way to also generate proper aliases for backwards compatibility
+# and also add the aliases to documentation!
+
     # Create a module for ourselves
-    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True, required_if=[['state', 'present', ['email']]])
+    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
     # Extract our parameters
-    state = module.params.get('state')
+    username = module.params.get('username')
+    if username and len(username) > 150:
```

Comment:
I would prefer not to check length here, it sounds like the API will error if you submit a value that's too long. If we check the length in our collection, we have to make sure those line up with the API (if the API ever changes), and I would prefer to avoid that type of maintanence